src/projects/projects.controller.ts

```diff
@@ -1,48 +1,78 @@
-import { Body, Controller, Delete, Get, HttpCode, Param, Post } from '@nestjs/common';
+import { BadRequestException, Body, Controller, Delete, Get, Param, Post, Query, UseGuards, UsePipes, ValidationPipe } from '@nestjs/common';
 import { AdminOnly } from 'src/auth/decorators/adminonly.decorator';
 import { CreateProjectDto } from './dto/create-project.dto';
 import { ProjectsService } from './projects.service';
 import { SanitizedProject } from './types/sanitized-project.type';
+import { RepositoryRequest, RepositoryResponse } from 'paastech-proto/types/proto/git-repo-manager';
+import { AuthGuard } from 'src/auth/guards/auth.guard';
+import { GetUser } from 'src/auth/decorators/user.decorator';
+import { GitRepoManagerService } from './git-repo-manager.service';
+import { UserDecoratorType } from 'src/auth/types/user-decorator.type';
+import { DeleteRepositoryDto } from './dto/delete-repository.dto';
 
 @Controller('projects')
 export class ProjectsController {
-    constructor(private projectsService: ProjectsService) {}
-
-    // GET /projects
-    // This action returns all of the authenticated user's projects
-    @AdminOnly()
-    @Get()
-    @HttpCode(501)
-    async findAll() {
-        // TODO: implement and remove the 501 decorator above
-        return '501 Not Implemented';
-    }
+  constructor(private projectsService: ProjectsService, private gitRepoManagerService: GitRepoManagerService) {}
 
-    // GET /projects/:uuid
-    // This action returns a #${id} project;
-    @Get(':uuid')
-    async findOne(@Param('uuid') id: string): Promise<SanitizedProject> {
-        return this.projectsService.findOne(id);
-    }
+  // GET /projects
+  // This action returns all of the authenticated user's projects
+  @Get()
+  async findAll(@GetUser() user: UserDecoratorType) {
+    return this.projectsService.findAll(user.id);
+  }
+
+  // GET /projects/:uuid
+  // This action returns a #${id} project;
+  @Get(':uuid')
+  async findOne(@Param('uuid') id: string, @GetUser() user: UserDecoratorType): Promise<SanitizedProject> {
+    return this.projectsService.findOne(id, user.id);
+  }
 
-    // POST /projects
-    /*
+  // POST /projects
+  /*
         This action creates a project
         and returns a NewProject Object
         with the project's id and name
     */
-    @Post()
-    @HttpCode(501)
-    async create(@Body() createProjectDto: CreateProjectDto) {
-        // TODO: implement and remove the 501 decorator
-        return '501 Not Implemented';
+  @Post()
+  @UsePipes(new ValidationPipe())
```

Comment:
`ValidationPipe` is set as global, see `main.ts` file for confirmation

Reply:
nice, done :+1: